grails-app/controllers/org/pih/warehouse/core/LocationController.groovy

```diff
@@ -83,6 +83,11 @@ class LocationController {
                 }
             }
 
+            Organization currentOrganization = locationInstance.organization
+            Organization pickedOrganization = Organization.get(params.organization?.id)
+            if (pickedOrganization && (pickedOrganization?.id != currentOrganization?.id) && !pickedOrganization?.active) {
```

Comment:
What if someone updates a location that has an organization that was deactivated before the location update? Should we also throw this exception to the user? In this version, this won't happen. Perhaps it is enough to do:
```
Organization organization = Organization.get(params.organization?.id)
if (organization && !organization.active) { throw exception }
```

Also - what if at this point there is no `params.organization?.id` and in line 96, the `findOrCreateSupplierOrganization` will find an inactive organization?

Reply:
> What if someone updates a location that has an organization that was deactivated before the location update? 
@awalkowiak I actually didn't want to it to cause an error, because a user might be confused when he/she goes to edit page, but decides to leave it as it is and just presses `Update` (but doesn't change the current organization. In my opinion it would be natural instinct to press `Update` as long as we don't have `Go back` or something like this button) and suddenly gets an error.
I think there would be no consequence if we allow a location to have an inactive organization (not while editing, but just by making already associated organization inactive), but at the same time we would punish/force the user to change the organization if he/she by mistake clicks `Edit Location`.
I just wanted to make sure, that if user doesn't change the organization while updating the location (which in the mean time could be set as inactive) and just keeps the 'old` organization, it would not cause the error.

That's my thought, I don't say it's the best idea or approach, it was just my point of view and that's why the code looks like this - it was done intentionally.
Let me know if I should leave it as it is or change to your approach and version.

Reply:
Make sure Katyrzyna reads these comments so she knows about some of the edge cases we've been discussing.